### PR TITLE
Remove xfail from passing grdimage tests

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

This PR removes the xfail mark from `test_grdimage_grid_and_shading_with_xarray` and splits off the passing tests from `test_grdimage_central_meridians_and_standard_parallels`, which marks all tests using **-JQ** as xfail. A suggestion for a more clever way to do this with pytest parameterize would be welcome.

Part of #1644
Closes https://github.com/GenericMappingTools/pygmt/issues/618

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

File: pygmt/tests/test_grdimage.py [hunk 1]
```diff
@@ -206,9 +202,10 @@ def test_grdimage_central_meridians(grid, proj_type, lon0):
 # are still slightly different with an RMS error of 25, see issue at
 # https://github.com/GenericMappingTools/pygmt/issues/390
 # TO-DO remove tol=1.5 and pytest.mark.xfail once bug is solved in upstream GMT
+#       combine with test_grdimage_central_meridians_and_standard_parallels_fix
 @check_figures_equal(tol=1.5)
```
Comment (seisman) [inline pygmt/tests/test_grdimage.py:206]: Can we remove the `tol=1.5` with GMT 6.3?

Comment (meghanrjones) [reply pygmt/tests/test_grdimage.py]: Yes, it's now removed.

Comment (weiji14) [reply pygmt/tests/test_grdimage.py]: Err, I think the `@check_figures_equal` decorator needs to be kept 😅

Comment (meghanrjones) [reply pygmt/tests/test_grdimage.py]: Wow, the dangers of 'just one more task' at the end of a workday 😅. I've added the decorator back in, so we can find out if the `tol` can be removed.

File: pygmt/tests/test_grdimage.py [hunk 2]
```diff
@@ -205,10 +201,11 @@ def test_grdimage_central_meridians(grid, proj_type, lon0):
 # Cylindrical Equidistant (Q) projections plotted with xarray and NetCDF grids
 # are still slightly different with an RMS error of 25, see issue at
 # https://github.com/GenericMappingTools/pygmt/issues/390
-# TO-DO remove tol=1.5 and pytest.mark.xfail once bug is solved in upstream GMT
-@check_figures_equal(tol=1.5)
+# TO-DO remove pytest.mark.xfail once bug is solved in upstream GMT
+#       combine with test_grdimage_central_meridians_and_standard_parallels_fix
+@check_figures_equal()
```
Comment (weiji14) [inline pygmt/tests/test_grdimage.py:206]: Need to put back the tolerance I think, still some small diffs resulting in test failures for the General Stereographic (S) projection plots


```python-traceback
FAILED pygmt/tests/test_grdimage.py::test_grdimage_central_meridians_and_standard_parallels[S-0-0-png] - pygmt.exceptions.GMTImageComparisonFailure: images not close...
FAILED pygmt/tests/test_grdimage.py::test_grdimage_central_meridians_and_standard_parallels[S-0-30-png] - pygmt.exceptions.GMTImageComparisonFailure: images not clos...
FAILED pygmt/tests/test_grdimage.py::test_grdimage_central_meridians_and_standard_parallels[S-123-30-png] - pygmt.exceptions.GMTImageComparisonFailure: images not cl...
FAILED pygmt/tests/test_grdimage.py::test_grdimage_central_meridians_and_standard_parallels[S-180-30-png] - pygmt.exceptions.GMTImageComparisonFailure: images not cl...
```

```suggestion
# TO-DO remove tol=1.5 and pytest.mark.xfail once bug is solved in upstream GMT
@check_figures_equal(tol=1.5)
```

File: pygmt/tests/test_grdimage.py [hunk 3]
```diff
@@ -205,10 +201,11 @@ def test_grdimage_central_meridians(grid, proj_type, lon0):
...
 @pytest.mark.parametrize("lat0", [0, 30])
-@pytest.mark.parametrize("lon0", [0, 123, 180])
+@pytest.mark.parametrize("lon0", [123, 180])
 @pytest.mark.parametrize("proj_type", [pytest.param("Q", marks=pytest.mark.xfail), "S"])
```
Comment (weiji14) [inline pygmt/tests/test_grdimage.py:209]: > A suggestion for a more clever way to do this with pytest parameterize would be welcome.

A lot more lines, but saves us from having to create another (very similar) unit test.

```suggestion
@pytest.mark.parametrize(
    ("proj_type", "lon0"),
    [
        ("Q", 0),
        pytest.param("Q", 123, marks=pytest.mark.xfail),
        pytest.param("Q", 180, marks=pytest.mark.xfail),
        ("S", 0),
        ("S", 123),
        ("S", 180),
    ],
)
```

File: pygmt/tests/test_grdimage.py [hunk 4]
```diff
@@ -222,3 +219,20 @@ def test_grdimage_central_meridians_and_standard_parallels(grid, proj_type, lon0
...
+    Test that plotting a grid with different central meridians (lon0) and
+    standard_parallels (lat0) using Cylindrical Equidistant (Q) and General
+    Stereographic (S) projection systems work.
+    """
+    fig_ref, fig_test = Figure(), Figure()
+    fig_ref.grdimage(
+        "@earth_relief_01d_g", projection=f"{proj_type}0/{lat0}/15c", cmap="geo"
+    )
+    fig_test.grdimage(grid, projection=f"{proj_type}0/{lat0}/15c", cmap="geo")
+    return fig_ref, fig_test
```
Comment (weiji14) [inline pygmt/tests/test_grdimage.py:238]:
```suggestion
```

**Review by seisman — APPROVED** [review]

**Pull request merged**